and-cli-deploy-azure-web-app.js

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env node
+require("./command-runner").run(async () => {
+    // -----------------------------------------------------------------------------------------
+    // #region Imports
+    // -----------------------------------------------------------------------------------------
+
+    const azure         = require("./_modules/azure");
+    const dotnetPath    = require("./_modules/dotnet-path");
+    const echo          = require("./_modules/echo");
+    const program       = require("commander");
+    const shell         = require("shelljs");
+
+    // #endregion Imports
+
+    // -----------------------------------------------------------------------------------------
+    // #region Variables
+    // -----------------------------------------------------------------------------------------
+
+    let   appName             = null;
+    let   branch              = null;
+    let   clientId            = null;
+    let   force               = false;
+    const pythonInstallerUrl  = "https://www.python.org/ftp/python/3.7.4/python-3.7.4-amd64.exe";
+    let   remote              = null;
+    let   resourceGroup       = null;
+    let   secret              = null;
+    let   tenantId            = null;
+    let   username            = null;
+
+    // #endregion Variables
+
+    /// -----------------------------------------------------------------------------------------
+    // #region Functions
+    // -----------------------------------------------------------------------------------------
+
+    const deployAzureWebApp = {
+        createRemoteIfMissing() {
+            if (shell.exec(`git remote get-url ${remote}`).code !== 0) {
+                const url = shell.exec(`az webapp deployment list-publishing-credentials --name ${appName} --resource-group ${resourceGroup} --query scmUri --output tsv`);
+
+                if (shell.exec(`git remote add ${remote} ${url}`).code !== 0)
+                {
+                    echo.error("Error trying to add remote!");
+                    azure.logout();
+                    shell.exit(1);
+                }
+            }
+        },
+        pushToRemote() {
```

Comment:
Would be awesome to see a `git` module scaffolded out with some of these, but not required right now 👍